### ISSUE TITLE
Errors on subscription renewal are not caught

### ISSUE THREAD

**Issue by foxxyz** [velocityzen/node-upnp#16]
### What Happens

If something goes wrong during the [renewal of a subscription](https://github.com/velocityzen/node-upnp/blob/f939055f105ca02dc2db7275819696bd562a6229/index.js#L258) triggered by a [`setTimeout`](https://github.com/velocityzen/node-upnp/blob/f939055f105ca02dc2db7275819696bd562a6229/index.js#L282), the ensuing error will not be caught and crashes the program.

```
/path/to/node_modules/node-upnp/error.js:50
  const err = new Error('Subscription renewal error');
              ^

Error: Subscription renewal error
    at Object.SubscriptionRenewal (/path/to/node_modules/node-upnp/error.js:50:15)
    at UPnPClient.renewSubscription (/path/to/node_modules/node-upnp/index.js:272:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'SUBSCRIBE_RENEW',
  statusCode: 412
}

Node.js v18.9.1
```

### What I Expected to Happen

I would expect it to keep retrying to renew the subscription in case of failure.

Curious on your thoughts?

**Comment by foxxyz** [comment]
@velocityzen is this intended behavior? It seems like this error should by caught in a `try/catch` within the [`setTimeout(...)` block](https://github.com/velocityzen/node-upnp/blob/f939055f105ca02dc2db7275819696bd562a6229/index.js#L282)

Would you accept a PR for this?

**Comment by foxxyz** [comment]
@velocityzen are you still maintaining this repository?

**Comment by velocityzen** [comment]
@foxxyz  sorry just noticed this. I do accept PRs!

**Comment by velocityzen** [comment]
@foxxyz  thank you. released in 1.3.0

**Comment by foxxyz** [comment]
Thank you!
